peps/pep-0791.rst

```diff
@@ -0,0 +1,165 @@
+to the mathematical functions defined by the C standard."  But as a state of
+art, over time the module was populated with functions that aren't related to
+the C standard or floating-point arithmetics.  Now it's much harder to describe
+module scope, content and interfaces (returned values or accepted arguments).
+
+For example, the :external+py3.14:mod:`math` module documentation says: "Except
+when explicitly noted otherwise, all return values are floats."  This is not
+longer true:  *None* of the functions listed in the "Number-theoretic
+functions" [1]_ subsection of the documentation returns a float, but the
+documentation doesn't say so.  In the proposed module a similar sentence "All
+return values are integers." could tell the truth once.  In a similar way we
+can simplify description of accepted arguments for both the
+:external+py3.14:mod:`math` and the new module.
+
+Apparently, the :external+py3.14:mod:`math` can't serve as a catch-all place
+for mathematical functions: we have also the :external+py3.14:mod:`cmath` and
+the :external+py3.14:mod:`statistics`.  Let's make same for integer-related
+functions.  It would provide shared context, which reduces verbosity in the
+documentation and conceptual load.  It also aids discoverability through
+grouping related functions and IDEs suggesting helpful completions.
+
+
+Specification
+=============
+
+The PEP proposes moving the following integer-related functions [1]_ in a new
+module, called ``imath``:
+
+* :external+py3.14:func:`~math.comb`
+* :external+py3.14:func:`~math.factorial`
+* :external+py3.14:func:`~math.gcd`
+* :external+py3.14:func:`~math.isqrt`
+* :external+py3.14:func:`~math.lcm`
+* :external+py3.14:func:`~math.perm`
+
+Their aliases in :external+py3.14:mod:`math` will be :term:`soft deprecated`.
+
+Modules functions will accept integers and objects that implement the
+:external+py3.14:meth:`~object.__index__` method which is used to convert the
+object to an integer number.
+
+Possible extensions for the new module and its scope are discussed in the
+`Open Issues <Open Issues_>`_ section.  New functions are not part of the
+proposal.
+
+
+Backwards Compatibility
+=======================
+
+As aliases in :external+py3.14:mod:`math` will be kept for an indefinite time
+(their use would be discouraged), there are no anticipated code breaks.
+
+
+Reference Implementation
+========================
+
+https://github.com/python/cpython/pull/133909
+
+
+Open Issues
+===========
+
+Module name
+-----------
+
+Chosen name seems consistent with other domain-specific mathematical module:
+:external+py3.14:mod:`cmath` (for complex numbers).
+
+There is already an ``imath`` project on PyPI, but only with two releases, with
+the most recent one four years ago.  Its repository is no longer accessible.
+The `Imath <https://github.com/AcademySoftwareFoundation/Imath>`_ C++ library
+include Python bindings with the same name.
```

Comment:
the way this `Imath` is presented here (grouped together with the unmaintained PyPI `imath`) might make it seem like the `Imath` one is also unmaintained / low-usage.
I'm sure it's not intentional - can we make it clearer that this other `Imath` is maintained and has real world usage?

Reply:
I did some rearrangement.